server/mqtt.go

```diff
@@ -3703,8 +4091,31 @@ func mqttDeliverMsgCbQos1(sub *subscription, pc *client, _ *Account, subject, re
 	}
 
 	topic := natsSubjectToMQTTTopic(strippedSubj)
+	retained := false // <>/<>
+	pc.mqttEnqueuePublishMsgTo(cc, sub, pi, qos, dup, retained, topic, msg)
+}
+
+func mqttDeliverPubRelCb(sub *subscription, pc *client, _ *Account, subject, reply string, rmsg []byte) {
+	hdr, _ := pc.msgParts(rmsg)
+	h := mqttParseInternalNATSHeader(hdr)
+
+	if h == nil || h.packetType != mqttPacketPubRel || h.pi == 0 {
+		return
+	}
 
-	pc.mqttDeliver(cc, sub, pi, dup, retained, topic, msg)
+	// This is the client associated with the subscription.
+	cc := sub.client
+
+	// This is immutable
+	sess := cc.mqtt.sess
+
+	sess.mu.Lock()
+	if sess.c != cc {
+		sess.mu.Unlock()
+		return
+	}
+
+	pc.mqttEnqueuePubRelTo(cc, h.pi)
```

Comment:
Unlock session, but then here you should have say `cc.mqttEnqueuePubRel(h.pi)`. We don't need addToPCD and therefore we don't need `pc` here. See comment in existing mqttEnqueuePubRel for details.